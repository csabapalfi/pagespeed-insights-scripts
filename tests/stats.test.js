const stats = require('../lib/stats');

describe('stats', () => {
  const mockSample = (data) => ({
    name: 'run 1', ...data
  });

  const samples = [
    mockSample({score: 80, FCP: 1, SI: 2, LCP: 3, TTI: 6, TBT: 5, CLS: 0.1}),
    mockSample({score: 90, FCP: 2, SI: 3, LCP: 4, TTI: 7, TBT: 6, CLS: 0.0}),
  ]

  it('returns expected stats', () => {
    expect(stats(samples))
      .toMatchSnapshot();
  });
});