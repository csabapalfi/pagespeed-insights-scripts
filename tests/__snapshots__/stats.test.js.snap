// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`stats returns expected stats 1`] = `
Array [
  Object {
    "CLS": 0.05,
    "FCP": 1.5,
    "LCP": 3.5,
    "SI": 2.5,
    "TBT": 5.5,
    "TTI": 6.5,
    "name": "median",
    "score": 85,
  },
  Object {
    "CLS": 0.07071067811865477,
    "FCP": 0.7071067811865476,
    "LCP": 0.7071067811865476,
    "SI": 0.7071067811865476,
    "TBT": 0.7071067811865476,
    "TTI": 0.7071067811865476,
    "name": "stddev",
    "score": 7.0710678118654755,
  },
  Object {
    "CLS": 0,
    "FCP": 1,
    "LCP": 3,
    "SI": 2,
    "TBT": 5,
    "TTI": 6,
    "name": "min",
    "score": 80,
  },
  Object {
    "CLS": 0.1,
    "FCP": 2,
    "LCP": 4,
    "SI": 3,
    "TBT": 6,
    "TTI": 7,
    "name": "max",
    "score": 90,
  },
]
`;
